software_training/src/action.cpp

```diff
@@ -0,0 +1,50 @@
+#include <ros/ros.h>
+#include <actionlib/client/simple_action_client.h>
+#include <actionlib>
+#include <std_msgs/String.h>
+#include <geometry_msgs/Twist.h>
+#include <sstream>
+#include <cmath>
+#include "software_training/Distance.h"
+#include <actionlib/server/simple_action_server.h>
+
+int main (int argc, char **argv)
+{
+  ros::init(argc, argv, "action");
+  
+  ros::NodeHandle nh;
+  ros::Publisher pub
+       = nh.advertise<geometry_msgs::Twist>("/moving_turtle/cmd_vel", 1000);
+  // create the action client
+  // true causes the client to spin its own thread
+  
+  //actionlib::SimpleActionClient<goemetry_msgs::Twist>("action", true);
+  actionlib::SimpleActionClient<software_training_assignment::Action> actionClient("action", true);
+
+  Server server(n, "action", boost::bind(&execute, _1, &server), false);
+  server.start();
+
+  ROS_INFO("Waiting for action server to start.");
+  // wait for the action server to start
+  actionClient.waitForServer(); //will wait for infinite time
+
+  ROS_INFO("Action server started, sending goal.");
+  // send a goal to the action
+  actionlib_tutorials::FibonacciGoal goal;
+  goal.order = 20;
+  ac.sendGoal(goal);
```

Comment:
it looks like you didn't build this because it does not compile, this should be actionClient. the client sends the goal to the server. make sure to run catkin build so that you can catch these mistakes in the code